apps/web/src/app/core/guards/has-premium.guard.ts

```diff
@@ -24,8 +25,14 @@ export function hasPremiumGuard(): CanActivateFn {
     const router = inject(Router);
     const messagingService = inject(MessagingService);
     const billingAccountProfileStateService = inject(BillingAccountProfileStateService);
+    const accountService = inject(AccountService);
 
-    return billingAccountProfileStateService.hasPremiumFromAnySource$.pipe(
+    return accountService.activeAccount$.pipe(
+      switchMap((account) =>
+        account
+          ? billingAccountProfileStateService.hasPremiumFromAnySource$(account.id)
+          : of(false),
+      ),
```

Comment:
No need to do this on this PR but I worry that some of the routes might navigate to a route protected with this right after logging in. Since active user state wouldn't emit until an account is set as active it would have delayed until that happened. I think we should give it a grace period to allow the account to come in and if it doesn't happen in a reasonable time throw an error because I think it would likely be a developer error. 

```suggestion
      filter((account) => account != null),
      timeout({
        first: 2000,
        with: () => {
          throw new Error("An active user is expected when checking if a user has premium.");
        },
      }),
      switchMap((account) =>
        billingAccountProfileStateService.hasPremiumFromAnySource$(account.id),
      ),
```

Please don't worry about doing this here though, you've got a lot of approvals and I can throw up a small PR doing it after you merge.